rewrite-core/src/main/java/org/openrewrite/internal/NameCaseConvention.java

```diff
@@ -149,18 +150,31 @@ private static String lowerUnderscore(String str) {
     private static String toCamelCase(String str, boolean lowerCaseFirstLetter) {
         boolean allUpperCase = true;
         final int strLength = str.length();
+
+        boolean uppercaseAbbreviationIncluded = false;
+        int uppercaseAbbreviationEndIndex = -1;
         for (int i = 0; i < strLength; i++) {
             final char c = str.charAt(i);
             if (Character.isLowerCase(c)) {
                 allUpperCase = false;
                 break;
             }
+
+            if ((i + 1) >= uppercaseAbbreviationMinLength) {
+                uppercaseAbbreviationIncluded = true;
+                uppercaseAbbreviationEndIndex = i - 1;
+            }
         }
```

Comment:
The check for `i` needs to be done at the end of checking. Thus, following proposal:

```suggestion
        int i = 0;
        while (i < strLength) {
            final char c = str.charAt(i);
            if (Character.isLowerCase(c)) {
                allUpperCase = false;
                break;
            }
            i++;
        }
        if (i > uppercaseAbbreviationMinLength) {
            uppercaseAbbreviationIncluded = true;
            uppercaseAbbreviationEndIndex = i - 1;
        }
```

Note that I also changed the if condition. I got rid of `+ 1` by using another comparison.

Reply:
Great idea. Let's go for that 👍